src/libs/ReportUtils.ts

```diff
@@ -1660,6 +1660,14 @@ function canAddOrDeleteTransactions(moneyRequestReport: OnyxEntry<Report>): bool
         return false;
     }
 
+    if (isIOUReport(moneyRequestReport) && currentUserAccountID !== moneyRequestReport?.managerID && currentUserAccountID !== moneyRequestReport?.ownerAccountID) {
+        return false;
+    }
+
+    if (isExpenseReport(moneyRequestReport) && currentUserAccountID !== moneyRequestReport?.managerID) {
```

Comment:
Wouldn't this break Instant Submit report where submitters create an expense and it gets added to a processing report?

Reply:
Yeah this breaks optimistic data for instant submit, we end up creating a new stale preview, but the backend adds the expense to the processing report


Uploading Screen Recording 2024-08-16 at 22.24.26.mov…